Doc/library/os.rst

```diff
@@ -569,6 +569,39 @@ process and user.
       See the documentation for :func:`getgroups` for cases where it may not
       return the same group list set by calling setgroups().
 
+.. function:: setns(fd, nstype=0)
+
+   Reassociate the current thread with a Linux namespace.
+   See the :manpage:`setns(2)` and :manpage:`namespaces(7)` man pages for more
+   details.
+
+   If *fd* refers to a :file:`/proc/{pid}/ns/` link, ``setns()`` reassociates the
+   calling thread with the namespace associated with that link, subject to any
+   constraints imposed by the *nstype* argument (a *nstype* of ``0`` means no
+   constraints).
+   Since Linux 5.8, *fd* may refer to a PID file descriptor obtained from
+   :func:`~os.pidfd_open`. In this case, ``setns()`` reassociates the calling thread
+   into one or more of the same namespaces as the thread referred to by *fd*
+   subject to any constraints imposed by *nstype*, which is a bit mask specified
+   by combining one or more of the ``CLONE_NEW*`` constants,
+   e.g., as ``setns(fd, os.CLONE_NEWUTS | os.CLONE_NEWPID)``.
+   The caller's memberships in unspecified namespaces are left unchanged.
+   *fd* can be any object with a :meth:`~io.IOBase.fileno` method, or a raw file descriptor.
```

Comment:
Are these sentences specific to the second case (I'm guessing maybe yes?), or do one or both apply to either? It is left unclear by the previous structure; as it stands being part of the newly-split second paragraph they apply to only the 
latter, but if one or both applies to both, then they should be separated into their own paragraph:

```suggestion

   The caller's memberships in unspecified namespaces are left unchanged.
   *fd* can be any object with a :meth:`~io.IOBase.fileno` method, or a raw file descriptor.
```

or 

```suggestion
   The caller's memberships in unspecified namespaces are left unchanged.

   *fd* can be any object with a :meth:`~io.IOBase.fileno` method, or a raw file descriptor.
```

If they both apply only to the second case, then this can be left as-is.